test/jdk/java/awt/FileDialog/FileNameOverrideTest.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright (c) 2011, 2024, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+import java.awt.FileDialog;
+import java.awt.Frame;
+import java.awt.Toolkit;
+import java.io.File;
+import java.io.IOException;
+import java.lang.reflect.InvocationTargetException;
+import javax.swing.JButton;
+import jtreg.SkippedException;
+
+/*
+ * @test
+ * @bug 6260659
+ * @summary File Name set programmatically in FileDialog is overridden during navigation, XToolkit
+ * @requires (os.family == "linux")
+ * @library /java/awt/regtesthelpers
+ * @library /test/lib
+ * @build PassFailJFrame
+ * @build jtreg.SkippedException
+ * @run main/manual FileNameOverrideTest
+ */
+
+public class FileNameOverrideTest {
+    private final static String fileName = "input";
+    private final static String clickDirName = "Directory for double click";
+    private final static String dirPath = System.getProperty("user.dir");;
+    private static JButton showBtn;
+    private static FileDialog fd;
+
+    private static JButton initialize() {
+        showBtn = new JButton("Show File Dialog");
+        showBtn.addActionListener(w -> {
+            Frame frame = new Frame();
+            fd = new FileDialog(frame, "Open");
```

Comment:
```suggestion
            fd = new FileDialog((Frame) null, "Open");
```
There is no point in this frame.

Reply:
Ok.